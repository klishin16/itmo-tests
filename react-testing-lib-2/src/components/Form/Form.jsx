import React, {useEffect, useState} from 'react';
import {Box, Button, TextField, Typography} from "@mui/material";

const Form = () => {
    const [data, setData] = useState(null)
    useEffect(() => {
        setTimeout(() => {
            setData({})
        }, 100)
    }, [])

    const [toggle, setToggle] = useState(false);
    const [value, setValue] = useState('');

    const onClick = () => setToggle(prev => !prev);


    return (
        <Box sx={{
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'space-around',
            alignItems: 'flex-start',
            gap: 1,
            p: 1,
            backgroundColor: 'whitesmoke'
        }}>
            <Typography variant='h5'>Hello world</Typography>
            <Typography variant='h3' data-testid="value-elem">{value}</Typography>
            {toggle === true && <Box sx={{ p: 2, bgcolor: 'primary.main', borderRadius: 1, color: 'white' }} data-testid="toggle-item">toggle</Box>}

            {data && <div style={{color: 'red'}}>data</div>}

            <Button variant="contained" data-testid="toggle-btn" onClick={onClick}>Click me</Button>
            <TextField
                onChange={e => setValue(e.target.value)}
                type="text" placeholder="Input value edited....."
                variant="standard"
            />


        </Box>
    )
};

export default Form;
