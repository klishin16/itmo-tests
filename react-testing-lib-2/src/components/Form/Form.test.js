import {fireEvent, render, screen} from "@testing-library/react";
import Form from "./Form";
import userEvent from "@testing-library/user-event";

describe('form tests', function () {
    test('hello world test', () => {
        render(<Form/>);
        const helloWorldElement = screen.getByText(/Hello world/i);
        expect(helloWorldElement).toBeInTheDocument();
    })

    test('button test', () => {
        render(<Form/>);
        const btnElement = screen.getByRole('button');
        expect(btnElement).toBeInTheDocument();
    })

    test('input test', () => {
        render(<Form/>);
        const inputElement = screen.getByPlaceholderText(/input value/i);
        expect(inputElement).toBeInTheDocument();
        // eslint-disable-next-line testing-library/no-debugging-utils
        // screen.debug();
        // expect(inputElement).toMatchSnapshot();
    })

    test('missing element test', () => {
        render(<Form/>);
        const btnElement = screen.queryByRole(/hello2/i);
        expect(btnElement).toBeNull();
    })

    test('delayed appearance test', async () => {
        render(<Form/>);
        const dataElement = await screen.findByText(/data/i);
        expect(dataElement).toBeInTheDocument();
        expect(dataElement).toHaveStyle({color: 'red'})
    })

    test('toggle test', () => {
        render(<Form/>);
        const btn = screen.getByTestId('toggle-btn');
        expect(screen.queryByTestId('toggle-item')).toBeNull();
        fireEvent.click(btn);
        expect(screen.queryByTestId('toggle-item')).toBeInTheDocument();
        fireEvent.click(btn);
        expect(screen.queryByTestId('toggle-item')).toBeNull();
    })

    test('input event test', () => {
        render(<Form />);
        const input = screen.getByPlaceholderText(/input value/i);
        expect(screen.queryByTestId('value-elem')).toContainHTML('');

        userEvent.type(input, 'info');
        expect(screen.queryByTestId('value-elem')).toContainHTML('info');
    })
});
