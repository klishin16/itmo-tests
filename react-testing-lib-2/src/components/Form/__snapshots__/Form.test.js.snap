// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form tests input test 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
  id="mui-3"
  placeholder="Input value....."
  type="text"
  value=""
/>
`;
